classes/requests/post/class-truelayer-request-create-payment.php

```diff
@@ -71,7 +71,6 @@ protected function get_body() {
                 'address_line1' => $order->get_billing_address_1(),
                 'address_line2' => $order->get_billing_address_2(),
                 'city' => $order->get_billing_city(),
-                'state' => ! empty( $state = $order->get_billing_state() ) ? $state : 'NA',
```

Comment:
Now state will never be sent to you.

If you change the code in the if statement to be:
```
$body['user']['address'] = array(
	'address_line1' => $order->get_billing_address_1(),
	'address_line2' => $order->get_billing_address_2(),
	'city' => $order->get_billing_city(),
	'zip' => $order->get_billing_postcode(),
	'country_code' => $order->get_billing_country(),
);

$state = $order->get_billing_state();
if( ! empty( $state ) ) {
	$body['user']['address']['state'] = $state;
}
```

It should function as before when a state is entered, and skip it if its empty value, so empty string, false, null etc.

Reply:
Ah yes, this is a good suggestion. Thanks!